src/edu/duke/cs/osprey/restypes/InterResBondingTemplate.java

```diff
@@ -209,16 +171,6 @@ public boolean atomCanBondOtherRes(Atom atom) {
             return atom.name.equalsIgnoreCase("N") || atom.name.equalsIgnoreCase("C")
                     || atom.name.equalsIgnoreCase("SG");
         }
-
-		@Override
-		public boolean isInterResBondedForward(Residue res1, Residue res2) {
```

Comment:
I'm a little worried about this change though. Like I mentioned before, I don't think the AtomConnectivity cache knows know how to deal with cross-linked molecules (eg disulfide-bonded CYX residues). If you give it one, it might return wrong forcefield parameters to you and not complain about any errors. So this loud error here might be announcing the presence of a silent elsewhere in the code. If we want to take out this error here, we'd also have to do some testing to be sure CYX residues are handled correctly end-to-end.

Reply:
Do you mean [this area](https://github.com/donaldlab/OSPREY3/blob/50d692c3f57b3793f7aaa3b5b6da35e650cd6793/src/edu/duke/cs/osprey/restypes/InterResBondingTemplate.java#L156) or specifically the cache, i.e. the lookup in the cache might fail due to because of changes to the key?

Reply:
I mean, the ResPairCache is part cache, and part hash table. There's a lot of redundancy in the atom pairs data. ie, an interaction between pair of glycine residues will have the same forcefield parameters if the resides are at positions 1&5, 1&7, or 20&32, since all atoms pairs are non-bonded. So you can save space by storing only one copy of that data, and then hash all the instances of that interaction to the same hash entry.

The trouble with things like disulfide bonds is that crosslinking breaks that position independence assumption, so the hash function in the hash table would mix together hash entries it's not supposed to. To properly support crosslinkers, we'd have to change the hash table somehow.

Or just use newer code that doesn't have this problem.

Reply:
That's clearer and makes sense.  I'm going to merge now as this fixes a confirmed bug and make a note to investigate how this affects the ResPairCache.  Also making a note to remove redundancies per discussion below about methods returning doubles.

Reply:
Added the issue here: https://github.com/donaldlab/OSPREY3/issues/140